AppCenter/AppCenter/Internals/MSServiceNotificationDelegate.h

```diff
@@ -0,0 +1,17 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#import <Foundation/Foundation.h>
+
+@class MSAppCenter;
+
+@protocol MSServiceNotificationDelegate <NSObject>
+
+/**
+ * A callback that is called when an App Center service notification is received.
+ * @param appCenter The app center instance.
+ * @param notificationData The service notification data.
+ */
+- (void)appCenter:(MSAppCenter *)appCenter didReceiveServiceNotification:(NSDictionary<NSString *, NSString *> *)notificationData;
```

Comment:
Does it actually make sense to keep this method optional? If it makes sense we need to add a test for delegates that don't implement this method. If it doesn't make sense and we make it required we don't need to check `respondsToSelector` on delegates before calling this method on them.

Reply:
Made it required and removed respondsToSelector.